src/lib/paths.test.ts

```diff
@@ -0,0 +1,76 @@
+import { parseProjectRoute } from './paths'
+import * as path from 'path'
+describe('testing parseProjectRoute', () => {
+  it('should parse a project as a subpath of project dir', async () => {
+    // @ts-ignore
+    global.window = { electron: { path: path } }
```

Comment:
This doesn't get cleaned up between runs. I really don't like modifying `global`. It's easy to make mistakes like this. Here, it introduces test order dependence, and that's a best case scenario. I'd much prefer having abstraction functions. But if we really can't get around it, I think we need to move this to a `beforeEach()` and in an `afterEach()`, remove it from `global`. This way, if there's an exception, it still cleans up.

Reply:
I switched to just injecting path into the function, we can then remove onDesktop as a param as it basically just checks if it's electron, and we know it's electron when `window.electron.path` is defined.

Reply:
Ha, that's broken though, not working when I test it. Maybe back to beforeEach for now.

Reply:
> we can then remove onDesktop as a param as it basically just checks if it's electron, and we know it's electron when `window.electron.path` is defined.

I think this makes sense. Then add a local var like this.

```typescript
const isDesktop = Boolean(pathlib)
```

Reply:
False alarm, CI looks pretty good outside of the known flakes. I tested again locally after rebuilding everything and it worked there too.